src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/PostClaimOrchestrationHandler.java

```diff
@@ -134,14 +139,20 @@ public void representativeIssueHandler(RepresentedClaimCreatedEvent event) {
             Function<Claim, Claim> doUploadSealedClaim =
                 c -> uploadSealedClaimOperation.perform(c, authorisation, sealedClaim);
 
-            Supplier<Claim> updatedClaim = () ->
-                doUploadSealedClaim
-                    .andThen(c -> rpaOperation.perform(c, authorisation, sealedClaim))
-                    .andThen(c -> notifyStaffOperation.perform(c, authorisation, sealedClaim))
-                    .andThen(c -> notifyRepresentativeOperation.perform(c, event))
-                    .apply(claim);
-
-            claimService.updateClaimState(authorisation, updatedClaim.get(), ClaimState.OPEN);
+            Claim updatedClaim = doUploadSealedClaim
+                .andThen(c -> rpaOperation.perform(c, authorisation, sealedClaim))
+                .andThen(c -> notifyStaffOperation.perform(c, authorisation, sealedClaim))
+                .andThen(c -> notifyRepresentativeOperation.perform(c, event))
+                .apply(claim);
+
+            updatedClaim.getState()
+                .filter(state -> state == CREATE)
+                .ifPresent(state -> {
+                    claimService.updateClaimState(
+                        authorisation,
+                        updatedClaim,
+                        state);
+                });
```

Comment:
`.filter(state -> state == CREATE)` could be simplified to `.filter(Predicate.isEqual(CREATE))`
(static import on `isEqual` to further simplify)
`.ifPresent(...)` body doesn't need braces as it's a single statement.

Reply:
done